source/Makefile

```diff
@@ -1,40 +1,35 @@
 .PHONY: python-html-book python-pdf-book r-html-book r-pdf-book \
 	clean
 
+python-all: python-html-book python-pdf-book
+
+r-all: r-html-book r-pdf-book
+
+Rmd_sources := $(shell python -c "import yaml; print(' '.join(yaml.load(open('_quarto.yml.template'),Loader=yaml.FullLoader)['book']['chapters']))")
+
 vector_images := $(wildcard diagrams/*.svg)
 
 diagrams := $(vector_images:.svg=.png)
 
 %.png : %.svg
 	inkscape --export-area-drawing --export-png=$@ --export-dpi=300 $<
 
-set-version-%:
+set-version-%: _quarto.yml.template _variables.yml.template
 	../scripts/set_version.py $*
 
-html-book-%:
-	$(MAKE) set-version-$*
+../%-book/index.html: set-version-% $(Rmd_sources) $(diagrams)
 	quarto render . --to html
 
-pdf-book-%:
-	$(MAKE) set-version-$*
+../%-book/resampling-with-python.pdf: set-version-% $(Rmd_sources) $(diagrams)
 	quarto render . --to pdf
 
-all-book-%:
-	# Since 0.2.260, quarto renders all formats by default.
-	$(MAKE) set-version-$*
-	quarto render .
-
-python-html-book: $(diagrams) html-book-python
-
-python-pdf-book: $(diagrams) pdf-book-python
-
-python-all: $(diagrams) all-book-python
+python-html-book: ../python-book/index.html
 
-r-html-book: $(diagrams) html-book-r
+python-pdf-book: ../python-book/resampling-with-python.pdf
```

Comment:
Do we need these targets now?  Can't we just use the long form - `../python-book/resampling-with-python.pdf` ?

Reply:
It's quite a bit more convenient to type, so perhaps leave for now?